### PR TITLE
Rename `Argument.help_record` to `get_help_record` + fix typos

### PR DESCRIPTION
**Pull request by alexreg**
* Rename `Argument.help_record` to `Argument.get_help_record`, to bring the name into line with `Option.get_help_record` (in Click).
* Typos.
* <s>Handle `None` returned by `Argument.get_help_record`.</s>
* <s>Handle case of empty sequence passed to `text_rows` param of `HelpFormatter._get_row_sep_for`.</s>

**Edited by @janluke**

### CODE REVIEW

File: cloup/_commands.py [hunk 1]
```diff
@@ -164,7 +162,7 @@ def add_command(
     ) -> None:
         super().add_command(cmd, name, section, fallback_to_default_section)
         name = cast(str, cmd.name) if name is None else name
-        aliases = getattr(cmd, 'aliases', [])
+        aliases = getattr(cmd, 'aliases', None) or ()
```
Comment (janluke) [inline cloup/_commands.py]: I don't think this change is needed. `self.aliases` is always a `List[str]` (eventually empty) if the command is a `cloup.Command` or `None` if it's not.

File: cloup/_option_groups.py [hunk 2]
```diff
@@ -185,19 +185,25 @@ def get_ungrouped_options(self, ctx: click.Context) -> Sequence[click.Option]:
...
+            )
+        )
+        if not help_records:
+            return None
         return HelpSection(
             heading="Positional arguments",
-            definitions=[
-                self.get_argument_help_record(arg, ctx) for arg in self.arguments
-            ]
+            definitions=list(filter(None, help_records)),
```
Comment (janluke) [inline cloup/_option_groups.py]: Filtering not needed. I'll refactor this method anyway.

**Review by janluke — COMMENTED** [review]
Thank you very much for the PR! I'll merge this after making some changes.

File: cloup/_option_groups.py (hunk 2 above)
Comment (alexreg) [reply cloup/_option_groups.py]: The idea is that `get_argument_help_record` can return `None` (given the signature of `get_help_record`), so it does in fact need to handle this case. Since a downstream user can subclass `Argument` or `Option` an override the `get_help_record` method.

File: cloup/_commands.py (hunk 1 above)
Comment (alexreg) [reply cloup/_commands.py]: Yeah, I thought you might say this. :-) In my [fork of *typer*](https://github.com/alexreg/typer), I was allowing it to be a tuple, and this made things easier... but of course I don't blame you if you only want to allow lists.

**Comment by alexreg** [comment]
@janluke Thank you, I appreciate that.

File: cloup/_option_groups.py [hunk 3]
```diff
@@ -185,19 +185,25 @@ def get_ungrouped_options(self, ctx: click.Context) -> Sequence[click.Option]:
...
+    ) -> Optional[Tuple[str, str]]:
         if isinstance(arg, cloup.Argument):
-            return arg.help_record(ctx)
+            return arg.get_help_record(ctx)
         return arg.make_metavar(), ""
 
     def get_arguments_help_section(self, ctx: click.Context) -> Optional[HelpSection]:
         if not any(getattr(arg, "help", None) for arg in self.arguments):
             return None
+        help_records = list(
```
Comment (janluke) [inline cloup/_option_groups.py]: I understand you're just being consistent with click and the semantic of `get_help_record` but `get_argument_help_record` never returns `None` and can't imagine a reason for it to return `None`. Click uses this semantic to filter out arguments and to avoid showing options with `hidden = True`. A positional argument, on the other end, should never be hidden. 

Let me know if I'm missing something!

File: cloup/_option_groups.py (hunk 2 above)
Comment (janluke) [reply cloup/_option_groups.py]: Sorry, I meant you already filtered at line 197.

Nonetheless... can you think of a reason why one would want to hide a positional argument from the help or worse from the usage string? I can't and I actually feel like that is something I don't want to allow. If you can't think of anything, I'd just "forbid" implementers from doing it by changing the return type to `Tuple[str, str]` (which is still fully compatible with the superclass signature).

File: cloup/_option_groups.py (hunk 3 above)
Comment (alexreg) [reply cloup/_option_groups.py]: @janluke Ah, I see how you are doing it now. Fair enough. I think the problem was caused by Typer supporting hidden arguments. Which is pretty unnatural, but it still supports them! I'll have a look into just supporting that on the Typer end.

File: cloup/_sections.py [hunk 4]
```diff
@@ -122,13 +122,13 @@ def __init__(
         :param kwargs:
             keyword arguments forwarded to the next class in the MRO
         """
+        super().__init__(*args, commands=commands, **kwargs)  # type: ignore
```
Comment (janluke) [inline cloup/_sections.py]: This change probably makes sense but as innocent as it may look, it's a breaking change lol. I'm not sure why I decided to call super() at the end but this change should be done in a separate PR.

Comment (alexreg) [reply cloup/_sections.py]: @janluke Yeah I worried about this... However, this change doesn't cause any tests to fail. The reason I moved it is because it's necessary to set `self.chain` before the rest of the code runs — otherwise you can get an error about the `chain` attribute not existing, in some cases.

Comment (janluke) [reply cloup/_sections.py]: Can you open a separate PR with this explanation, please?

Comment (janluke) [reply cloup/_sections.py]: Opened an issue about it: 
- #118

File: cloup/_option_groups.py (hunk 3 above)
Comment (janluke) [reply cloup/_option_groups.py]: Great!

**Comment by alexreg** [comment]
@janluke Looks good. Are you going to basically cherry-pick everything in this PR except the bit you just created a new issue for (reordering of `super().__init__`) and possibly [this](https://github.com/janluke/cloup/pull/116#discussion_r873235075) (not sure)? If you prefer for me to revert the bits you don't want in this PR, I can do that.

**Comment by janluke** [comment]
@alexreg I had already reverted all changes I don't want to include but they were overwritten by your force-push. Sorry, I should have been clearer I would have made the changes. No need to change anything on your end!

I'll just keep the method renaming and typo fixes. All other things will be reverted (sorry!), i.e. the return type of `Argument.get_help_record` is going to be `Tuple[str, str]`

**Comment by alexreg** [comment]
@janluke Oops, sorry about that. And yes, no worries. You kept most of the choices. Given our discussion, I agree it makes no sense to change the type signature (just the name, like you did).

Not sure if you meant to change the formatting of imports, or whether it was just my formatter messing things up here, and it accidentally crept into the commit? This one, I mean.
```python3
from typing import (Any, Callable, Dict, Iterable, List, NamedTuple, Optional,
                     Tuple, Type, TypeVar, cast, overload)
```
```

**Comment by janluke** [comment]
@alexreg No problem, fixed in master
